desc/builder/field.go

```diff
@@ -508,6 +508,11 @@ func (flb *FieldBuilder) buildProto(path []int32, sourceInfo *dpb.SourceCodeInfo
 		def = proto.String(flb.Default)
 	}
 
+	// normal (non-messageset) messages have lower tag limit, so enforce that here
+	if !isMessageSet && flb.number > internal.MaxNormalTag {
```

Comment:
Do you want to enforce the `isMessageSet` case as well here? I'm sure you've thought of this and there's likely a reason you're not, but just checking.

Reply:
The code already validates that tags are never > `internal.MaxTag` when fields are created. This is a deferred check just for the tighter range imposed on non-messageset fields, which can't be performed until we know the field's context (e.g. is it part of a messageset or not).

Reply:
Ah ok

Reply:
I changed this to just use `internal.GetMaxTag` so it's simpler to follow along